### ISSUE TITLE
Constant "IntelliSenseMode was changed" warnings

### ISSUE THREAD

**Issue by krisj07** [microsoft/vscode-cpptools#8286]
Bug type: Language Service

**Describe the bug**
- OS and Version: Ubuntu 20.04
- VS Code Version: Version: 1.61.0
Commit: ee8c7def80afc00dd6e593ef12f37756d8f504ea
Date: 2021-10-07T18:11:02.929Z
Electron: 13.5.1
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.11.0-37-generic
- C/C++ Extension Version: 1.7.0

Every time VSCode is started the hidden output panel pops up with the "C/C++ Configuration Warnings" view which shows the following message:

```
[10/16/2021, 4:54:43 PM] For C source files, IntelliSenseMode was changed from "linux-clang-x64" to "linux-gcc-x64" based on compiler args and querying compilerPath: "/usr/bin/gcc"
[10/16/2021, 4:54:43 PM] For C++ source files, IntelliSenseMode was changed from "linux-clang-x64" to "linux-gcc-x64" based on compiler args and querying compilerPath: "/usr/bin/gcc"
```

Despite manually setting the IntelliSenseMode to "linux-gcc-x64" the warnings still appear.

**Steps to reproduce**

1. Install VSCode. 
2. Install C/C++ extension
3. Run VSCode
4. Manually set "C_Cpp.default.intelliSenseMode": "linux-gcc-x64"
5. Run VSCode again

**Expected behavior**

The C/C++ extension should read its defined IntelliSenseMode setting instead of continuously reset it "based on compiler args and querying compilerPath". After manually defining the IntelliSenseMode the warnings should not appear any more.

**Comment by Colengms** [comment]
Hi @krisj07 .  Thanks for reporting this.  It looks like the default `compilerPath` is taking precedence, rather than configuring without any compiler (which system include paths and system defines are retrieved from).  I believe you could work around the issue either by setting `compilerPath` to the matching compiler, or setting it to `""` to entirely prevent use of a compiler.

We'll use this issue to track either ignoring the compilerPath if it doesn't match an explicitly set IntelliSense Mode, or trying to choose a found compiler that matches the IntelliSenseMode.

**Comment by sean-mcmanus** [comment]
Yeah, if the intelliSenseMode setting is set to clang in c_cpp_properties.json, then that will take precedence over the settings.json, which is "by design". Can we close this issue?

**Comment by krisj07** [comment]
Yes, the issue was resolved, so I'll close it now.

Regarding the settings file, looking at my project history the 'c_cpp_properties.json' file had originally been initialized by the CPP extension itself. At least in my case what happened was that 1) the VSCode CPP extension made the 'c_cpp_properties.json' file, 2) the values in 'c_cpp_properties.json' are not visible in the VSCode settings view and cannot be changed there, but 3) the values in 'c_cpp_properties.json' take precedence over the visible VSCode settings.

Once I found out about the 'c_cpp_properties.json' file the problem was easy to fix, but since it cannot be fixed through VSCode settings, and there is no indication why the regular VSCode settings are being overridden, it can be a bit misleading. In any case, it's a very minor issue.
